### ISSUE TITLE
LongType.setReal(longNumber) sets the variable to Integer.MAX_VALUE

### ISSUE THREAD

**Issue by LeonYang5114** [imglib/imglib2#110]
When setting the real part of a LongType to a large number using setReal(), the value of that variable will be set to Integer.MAX_VALUE instead.
Code to generate the situation:
```
LongType num = new LongType();
num.setReal(9876543210L);    // number larger than 2147483647
System.out.println(num.get());    // outputs "2.147483647E9"
```
The issue is rooted in the fact that `setReal()` accepts either `float` or `double` as input, but when using a `long`, Java automatically selects the [`float` version](https://github.com/imglib/imglib2/blob/master/src/main/java/net/imglib2/type/numeric/integer/LongType.java#L177). After the `long` is rounded as a `float`, it is then [casted to an `int`](https://github.com/imglib/imglib2/blob/master/src/main/java/net/imglib2/util/Util.java#L487), which causes the problem.
We could just cast the `long` to a `double` manually to avoid this problem, but I think that's quite tricky. Removing the `setReal(float)`method could be a solution.

**Comment by tpietzsch** [comment]
@LeonYang5114 No, we cannot remove the `setReal(float)` method. It is defined further up in the hierarchy and for example for `FloatType` it makes a lot of sense. Any reason why you wouldn't use `setInteger()` instead?

**Comment by ctrueden** [comment]
> Any reason why you wouldn't use `setInteger()` instead?
If you have a `T extends RealType<T>` then `setInteger` isn't available. I think it is OK to write `num.setReal((double) myLong)`. Alternately, and perhaps better, would be to add a `setReal(long)` to the API contract of `RealType` so that specific type implementations can do something less lossy when it comes to `long`s.

**Comment by tpietzsch** [comment]
@ctrueden Yep, that is a good idea

**Comment by ctrueden** [comment]
Cool, @LeonYang5114 would you care to make a PR for that?
